src/HomePage/components/MoviesList/MoviesList.tsx

```diff
@@ -4,7 +4,33 @@ import './MoviesList.css'
 import {Movie} from "../../../Mocks";
 
 export function MoviesList({ movies } : any) {
+    const [state, setState] = React.useState({
+        isActionsPopupOpen: false,
+        isCardEditable: false,
+        isCardDeletable: false,
+        cardNumber: -1
+    })
+
     return (<div className='cards'>
-        {movies.map( (movie: Movie) => <MovieCard key={movie.original_title} title={movie.original_title} popularity={movie.popularity}/>)}
+        {movies.map( (movie: Movie, index: number) => <MovieCard
+            onMovieCardClick={() => setState(prevState => ({
+                ...prevState,
+                isActionsPopupOpen: !state.isActionsPopupOpen,
+                cardNumber: index
+            }))}
+            onMovieCardEditable={() => setState(prevState => ({
```

Comment:
NIT: would recommend to call onMovieEditClick, on MovieDeleteClick and onMovieClick - but this is not necessary